setup_hooks.sh

```diff
@@ -1,2 +1,2 @@
 #!/bin/bash
-ln -s $PWD/hooks/pre-push $PWD/.git/hooks/pre-push
+ln -s $PWD/doc/dev/hooks/pre-push $PWD/.git/hooks/pre-push
```

Comment:
Not sure how I feel about this being moved to `doc`, maybe `tools` or `scripts` is better?

Reply:
@ericl Agree with @edoakes, can you move it to `scripts/`?